src/lib.rs

```diff
@@ -1434,15 +1434,20 @@ impl<'a> Iterator for OutputIter<'a> {
 /// are available to allocate and what register may be used as a
 /// scratch register for each class, and some other miscellaneous info
 /// as well.
+///
+/// Note that `MachineEnv` is designed to be a global configuration struct that programs
+/// will have very few of and generally want to keep around for their entire lifetime.
+/// In order to make static initialization easier the registers lists are static slices instead
+/// of `Vec`s. If your use case depends on dynamically creating the registers lists, consider
+/// [`Vec::leak`].
```

Comment:
I'd prefer not to give this suggestion in the doc-comment: since it's parameterized on a lifetime (rather than hardcoding `'static`), it's possible to build a `MachineEnv` dynamically that is a borrowed view on owned `Vec`s, basically giving the same capabilities as before. This is fairly idiomatic in Rust, so I don't think we need a special note or recommendation.

Reply:
ah yeah good catch, that was from before I allowed any lifetimes. This notice doesn't make much sense anymore